File: go/math_activities.go
package hello

import "context"

// Arithmetic activities (the pipeline, non-determinism and versioned demos).

// Add returns a + b.
func Add(ctx context.Context, in MathInput) (int, error) {
	// >>> BREAKPOINT (activity) <<<
	return in.A + in.B, nil
}

// Double returns value * 2.
func Double(ctx context.Context, in DoubleInput) (int, error) {
	// >>> BREAKPOINT (activity) <<<
	return in.Value * 2, nil
}

// Square returns value * value.
func Square(ctx context.Context, in SquareInput) (int, error) {
	// >>> BREAKPOINT (activity) <<<
	return in.Value * in.Value, nil
}

File: python/math_activities.py
"""Arithmetic activities (the pipeline, non-determinism and versioned demos)."""

from temporalio import activity

from shared import MathInput, DoubleInput, SquareInput


@activity.defn
async def add(inp: MathInput) -> int:
    # >>> BREAKPOINT (activity) <<<
    return inp.a + inp.b


@activity.defn
async def double(inp: DoubleInput) -> int:
    # >>> BREAKPOINT (activity) <<<
    return inp.value * 2


@activity.defn
async def square(inp: SquareInput) -> int:
    # >>> BREAKPOINT (activity) <<<
    return inp.value * inp.value
